### ISSUE TITLE
Support for binary-only metapackages

### ISSUE THREAD

**Issue by wyphan** [fortran-lang/fpm#566]
### Description

Is it possible to have binary-only "metapackages" in fpm?
I think this might be useful for vendor-provided closed-source packages like IBM ESSL, Intel MKL, or AMD AOCL, which can then be used as a prerequisite to provide the optimized BLAS library for building LAPACK.

### Possible Solution

Add a new target type, perhaps?
Specifically for Intel MKL on Linux, I think it can be implemented using a Bash shell script that simply looks for the MKL root directory under `/opt/intel` or `/opt/intel/oneapi`.
A similar procedure can be done for AMD AOCL, where the library is installed under `/opt/amd`

### Additional Information

This is actually something that the one of the LAPACK developers informed me about: it is not recommended to build LAPACK using reference BLAS, as it is slow and not optimized.

**Comment by LKedward** [comment]
Thanks @wyphan and @zoziha - fpm was not originally envisaged to be a binary package manager (I think), because of fundamental incompatibilities between different compilers and different versions. As already [pointed out](https://github.com/fortran-lang/fpm/issues/670#issuecomment-1063005790), every combination of operating system + compiler + compiler version requires its own binary distribution to ensure compatibility. The binary packages would therefore only be useable by those who have a supported OS+compiler+compiler version combination. Instead, I would recommend linking against a binary package installed by your system package manager. Currently the main difficulty with this is correctly locating these system-installed packages (see #330).

Your proposed use-case of using optimised LAPACK is already possible with *fpm* by linking against a system-installed implementation. For example, to use OpenBLAS, you can add the following to your `fpm.toml`:
```toml
[build]
link = "openblas"
```

**Comment by zoziha** [comment]
I also agree that we should be careful about publishing binary-only fpm packages, which creates a certain complexity and compatibility.
Maybe I think it's more important:
How to make the package can be installed locally and easily linked by fpm, maybe we can create **a unique file path for fpm**, store these libraries, and **add it to the environment path by default**; fpm as a build system, linking with native libraries is required.
The effect of link='openblas' is because we can add the library to the `LIBRARY_PATH` environment variable, but the search of `.mod` files is also a problem (currently `external-modules` only supports relative paths).

Maybe we shouldn't publish binary-only fpm packages, but can add `fpm.toml` index for libraries installed locally by fpm to facilitate fpm parsing and calling?

#### Link
- [How to link locally built third party libraries (fpm discussion)](https://github.com/fortran-lang/fpm/discussions/477)
